app/data/views.py

```diff
@@ -331,57 +332,152 @@ def _query_param_to_case_stmnt(self, param, request, queryset):
+                'label': lambda x: calendar.month_name[x]
+            },
+            'week_of_year': {
+                'range': xrange(1, 54),  # Up to 53 weeks in a year
+                'lookup': lambda x: {'occurred_from__week': x},
+                'label': lambda x: 'Week {}'.format(str(x))
+            },
+            'day_of_week': {
+                'range': xrange(1, 8),
+                'lookup': lambda x: {'occurred_from__week_day': x},
+                'label': lambda x: calendar.day_name[x-1]
+            },
+            'day_of_month': {
+                'range': xrange(1, 32),
+                'lookup': lambda x: {'occurred_from__day': x},
+                'label': lambda x: str(x)
+            },
+            'hour_of_day': {
+                'range': xrange(0, 24),
+                'lookup': lambda x: {'occurred_from__hour': x},
+                'label': lambda x: '{}:00'.format(x)
+            },
         }
 
-        # TODO: This sets year range based on the oldest and youngest records in the filtered
-        # queryset, but it (and possibly other date ranges) should be defined by the request's
-        # occurred_max and occurred_min parameters if available.
-        if period_type == 'year':
-            range_min = queryset.order_by('occurred_from').first().occurred_from.year
-            range_max = queryset.order_by('-occurred_from').first().occurred_from.year + 1
-            period_range = xrange(range_min, range_max)
-            period_labels = lambda x: str(x)
-        elif period_type in easy_ranges.keys():
-            period_range = easy_ranges[period_type]['keys']
-            period_labels = easy_ranges[period_type]['labels']
+        sequential_ranges = {
+            'year': {
+                'range': [],
+                'lookup': lambda x: {'occurred_from__year': x},
+                'label': lambda x: str(x)
+            },
+            'month': {
+                'range': [],
+                'lookup': lambda (yr, mo): {'occurred_from__month': mo, 'occurred_from__year': yr},
+                'label': lambda (yr, mo): '{}, {}'.format(calendar.month_name[mo], str(yr))
+            },
+            'week': {
+                'range': [],
+                'lookup': lambda (yr, wk): {'occurred_from__week': wk, 'occurred_from__year': yr},
+                'label': lambda (yr, wk): '{} Week {}'.format(str(yr), str(wk))
+            },
+            'day': {
+                'range': [],
+                'lookup': lambda (yr, wk, day): {'occurred_from__week': wk,
+                                                 'occurred_from__year': yr,
+                                                 'occurred_from__day': day},
+                'label': lambda (yr, wk, day): template_date(datetime.date(yr, wk, day))
+            },
+        }
+
+        if period_type in easy_ranges.keys():
+            period = easy_ranges[period_type]
+        elif period_type in sequential_ranges.keys():
```

Comment:
Hmm, I feel like this could be a potential booby trap if someone modifies `easy_ranges` without realizing that it has to be disjoint from `sequential_ranges`. Not exactly sure what to do about it though or if it's even worth addressing since there aren't many periodic aggregations outside of what's already here.